### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by kahunacohen**
More emphasis on the package's flexibility and how it can allow you to slowly abstract your tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,78 @@
...
+# Robot Framework Page Objects 
+## Installing
 
-This Python package adds support of the [Page Object](http://martinfowler.com/bliki/PageObject.html) pattern to [Robot Framework](http://robotframework.org/) & Robot Framework's [Selenium2Library](https://github.com/rtomac/robotframework-selenium2library).  
+    $ pip install robotframework-pageobjects
+## What it is 
+This Python package adds support for the [Page Object](http://martinfowler.com/bliki/PageObject.html) pattern with [Robot Framework](http://robotframework.org/) and Robot Framework's [Selenium2Library](https://github.com/rtomac/robotframework-selenium2library). Though this
+package is a [Robot library](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#extending-robot-framework), it's usable outside the Robot context and facilitates use of the page object pattern independently of any Python testing framework. This means you can use it to create
+page objects and run tests in other testing frameworks like  
+<a href="http://docs.python.org/2/library/unittest.html"><code>unittest</code></a>,
```
Comment (hellmanj) [inline README.md]: Two different kinds of markup links here is odd

File: README.md [hunk 2]
```diff
@@ -1,17 +1,78 @@
...
 
-## Installing
+        def tearDown(self):
+            self.page.close()
 
-    $ pip install robotframework-pageobjects
+        def test_title(self):
+            self.page.title_should_be("My Page")
+
+        def test_portlet_renders(self):
```
Comment (hellmanj) [inline README.md]: This method isn't used in the example.  I think your point would be clearer if it is.

File: README.md [hunk 3]
```diff
@@ -1,17 +1,78 @@
...
+            self.page.element_should_be_visible("the portlet")
+
+
+We could run this test on the default browser ([PhantomJs](http://phantomjs.org/)) like this:
+
+    $ export PO_BASEURL=http://qa.mydomain.com 
+    $ export PO_SELENIUM_SPEED=1 # Slow the whole test down for debugging
+    $ python mytest.py
```
Comment (hellmanj) [inline README.md]: I don't think this would actually work...

File: README.md [hunk 4]
```diff
@@ -1,17 +1,78 @@
...
+- Built-in and readable *assertion methods* on the page object. For example `location_should_be`, `title_should_be` etc.
+- Much more...
+
+Like we said, the package is very flexible: it doesn't force you to use Robot, nor forces you
+to do heavy page object modeling up front. This is great for convicing your organization to move toward
+BDD and page object development because you can approach those ideals iteratively. Even
+with light modeling, and use of a non-BDD framework, your test suites
+can still benefit from the above listed features. Here's an example of a very minimally
+abstracted page object, where we're using a page object assertion (`title_should_be`) 
+outside of the Robot framework. `mytest.py`
```
Comment (hellmanj) [inline README.md]: Is there a better way to state the title of the example file without it being on the same line as that paragraph?

File: README.md [hunk 5]
```diff
@@ -1,17 +1,78 @@
...
+Notice, we did not factor out all page implementation details from the test itself. But
+we still can leverage many of the package's features in our tests. If we need to model the page
+further, nothing stops us from doing so in the future. 
+We'll learn how to do this in a bit.
+
+## More on page objects
+The main point of using page objects is to factor out page implementation details (element locators, UI details etc.) from the actual test suites. This makes the tests read more about the services a page offers and what's being tested instead of the internals of the page. It also makes your tests much more maintainable. For example, if a developer changes an element ID, you only need make that change once--in the appropriate page object.
+
+## How it works
+Each page object you create is simply a Robot library that inherits from this package's base `Page` class. Since these classes are *plain old Python classes* they can work independently of Robot
```
Comment (hellmanj) [inline README.md]: To maintain the framework-agnostic image, I'd change "a Robot library" to "an object" in the first part of this sentence and then mention that in the context of robot, it becomes a "Robot library"

File: README.md (hunk 3 above)
Comment (kahunacohen) [inline README.md]: Meaning, we'd need to run `unttest.main()`, right?

Comment (hellmanj) [inline README.md]: Right, or use nose, etc

**Comment by kahunacohen** [comment]
OK @hellmanj , pushed up changes related to your comments.

**Comment by hellmanj** [comment]
:+1:

**Pull request merged**